src/_common/content/content-viewer/components/AppContentViewerBaseComponent.vue

```diff
@@ -53,23 +54,54 @@ function getComponentType(data: ContentObject): any {
 	}
 }
 
-export function renderChildren(childObjects: ReadonlyArray<ContentObject>) {
+export function renderChildren(
+	childObjects: ReadonlyArray<ContentObject>,
+	contentRules?: ContentRules
+) {
 	const children = [];
 	if (childObjects) {
+		let prevWasParagraph = false;
+		const inlineParagraphs = contentRules?.inlineParagraphs === true;
+
 		for (const contentData of childObjects) {
+			const currentIsParagraph = contentData.type === 'paragraph';
+
+			// Add a space of padding around each paragraph we're trying to
+			// render inline.
+			if (inlineParagraphs && (prevWasParagraph || currentIsParagraph)) {
```

Comment:
Can you make the paragraphs inline-block and just put some margin around them? Or does it not collapse the margins properly with our HTML structure?

Reply:
Inline-block doesn't work to collapse text in the way we want it here.